vendor/bat-native-confirmations/src/bat/confirmations/internal/payments.cc

```diff
@@ -177,10 +177,14 @@ base::Time Payments::CalculateNextPaymentDate(
     year++;
   }
 
-  auto date = base::StringPrintf("%d-%d-%d", year, month, kNextPaymentDay);
+  base::Time::Exploded next_payment_date_exploded = now_exploded;
+  next_payment_date_exploded.year = year;
+  next_payment_date_exploded.month = month;
+  next_payment_date_exploded.day_of_month = kNextPaymentDay;
 
   base::Time next_payment_date;
-  bool success = base::Time::FromString(date.c_str(), &next_payment_date);
```

Comment:
FromString was working as expected on desktop, however, on Android, the dates were shown incorrectly